src/main/java/seedu/address/logic/parser/HintParser.java

```diff
@@ -49,6 +52,72 @@
     private static String arguments;
     private static String userInput;
 
+    private static final Set<String> COMMAND_SET = new HashSet<String>(Arrays.asList(new String[]{
+        "add", "alias", "clear", "delete", "edit", "exit", "find", "help", "history", "list",
+        "music", "redo", "remark", "select", "unalias", "undo"
+    }));
+
+    /**
+     * Parses {@code String input} and returns an appropriate autocompletion
+     */
+    public static String autocomplete(String input) {
+        String[] command;
+
+        try {
+            command = ParserUtil.parseCommandAndArguments(input);
+        } catch (ParseException e) {
+            return "";
+        }
+
+        userInput = input;
+        commandWord = command[0];
+        arguments = command[1];
+
+        switch (commandWord) {
+
+        case AddCommand.COMMAND_WORD:
+            return input + generateAddHint();
+
+        case EditCommand.COMMAND_WORD:
+            return input + generateEditHint();
+
+        case FindCommand.COMMAND_WORD:
+            return input + generateFindHint();
+
+        case MusicCommand.COMMAND_WORD:
+            if (arguments.isEmpty()) {
+                return commandWord + " " + (MusicCommand.isPlaying() ? "pause" : "play");
+            }
+            return commandWord + " " + autocompleteFromList(arguments.trim(), new String[] {"play", "pause", "stop"});
+
+        default:
+            return autocompleteCommand(commandWord);
+        }
+    }
+
+    /**
+     * Parses {@code String command} and returns the closest matching command word.
+     */
+    public static String autocompleteCommand(String command) {
+        Set<String> commands = new HashSet<>();
+        commands.addAll(COMMAND_SET);
+        commands.addAll(UserPrefs.getInstance().getAliases().getAllAliases());
+        String[] list = commands.toArray(new String[commands.size()]);
+        return autocompleteFromList(command, list);
+    }
+
+    /**
+     * Parses {@code String input} and returns the closest matching string in {@code String[] strings}.
+     */
+    public static String autocompleteFromList(String input, String[] strings) {
```

Comment:
What are do strings refer to? or is this just a general util function?

Reply:
They are the strings we should autocomplete from. Should it be renamed to `needle` and `haystack` or what do you suggest?